### PR TITLE
A03 - Add git config mock

### PR DESCRIPTION
**Pull request by blaggacao**
Add git config mock and use ConfigurationInterface in all places, so it can be mocked (for testing)

This is preparative for #1483, where I need to patch `git.Config().GetGiteaToken()`

### CODE REVIEW

**Comment by blaggacao** [comment]
```shell
make unit
env GOGC=off go test -count=1 -timeout 60s -race ./src/... ./test/...
?   	github.com/git-town/git-town/src/browsers	[no test files]
?   	github.com/git-town/git-town/src/cfmt	[no test files]
?   	github.com/git-town/git-town/src/cmd	[no test files]
ok  	github.com/git-town/git-town/src/command	0.106s
ok  	github.com/git-town/git-town/src/drivers	0.078s
?   	github.com/git-town/git-town/src/dryrun	[no test files]
ok  	github.com/git-town/git-town/src/git	1.243s
?   	github.com/git-town/git-town/src/prompt	[no test files]
?   	github.com/git-town/git-town/src/script	[no test files]
ok  	github.com/git-town/git-town/src/steps	0.040s
?   	github.com/git-town/git-town/src/util	[no test files]
ok  	github.com/git-town/git-town/test	1.857s
ok  	github.com/git-town/git-town/test/helpers	0.029s
```

**Comment by blaggacao** [comment]
This is preparative for #1483, where I need to patch `git.Config().GetGiteaToken()`

**Comment by blaggacao** [comment]
Blocked for https://github.com/vektra/mockery/issues/272 to be resolved

**Comment by kevgo** [comment]
I've been reading through your changes for the last hour. I think what I describe in https://github.com/git-town/git-town/pull/1499#discussion_r434236553 is the best solution for mocking stuff here. I'm happy to implement this for the other drivers. What do you think?

File: src/git/config.go [hunk 1]
```diff
@@ -493,3 +550,6 @@ func (c *Configuration) ValidateIsOnline() error {
 	}
 	return nil
 }
+
+// Configuration implements ConfigurationInterface
+var _ ConfigurationInterface = (*Configuration)(nil)
```
[the remark this answers is not in the document]
Comment (blaggacao) [reply src/git/config.go:554]: It fails during build time if the interface is not met. Kind of obscure but that's how folks at k8s do it all over.

File: src/git/prod_repo.go [hunk 2]
```diff
@@ -4,10 +4,10 @@ import "github.com/git-town/git-town/src/command"
...
 // ProdRepo is a Git Repo in production code.
 type ProdRepo struct {
-	Silent         CachedRunner  // the Runner instance for silent Git operations
-	Logging        CachedRunner  // the Runner instance to Git operations that show up in the output
-	LoggingShell   *LoggingShell // the LoggingShell instance used
-	*Configuration               // the git.Configuration instance for this repo
+	Silent                 CachedRunner  // the Runner instance for silent Git operations
+	Logging                CachedRunner  // the Runner instance to Git operations that show up in the output
+	LoggingShell           *LoggingShell // the LoggingShell instance used
+	ConfigurationInterface               // the interface to an initialized Configuration instance for this repo
```
[the remark this answers is not in the document]
Comment (blaggacao) [reply src/git/prod_repo.go:10]: Sounds good!

**Comment by blaggacao** [comment]
I think implementing the relevant interface on the caller side is just a prefect, solution for #1498 
This is not required for #1499 

Thanks for the hint!

**Comment by blaggacao** [comment]
Superseded by https://github.com/git-town/git-town/pull/1508

**Pull request closed without merging**